### ISSUE TITLE
Add tests for users API

### ISSUE THREAD

**Issue by ericaddison** [ericaddison/Hailing_Frequencies#17]
There are good tests now for `Conversations`, `ConvMessages`, and `ConvUsers`, but not really for `Users`. Should be simple ... I just need to add them in.

**Comment by psigourney** [comment]
I'm confused about the logic behind the Users API.

1) The FirebaseHeader is calling /api/users/ and passing the firebase token. With the intent of authenticating the current user and retrieving the local user object from ndb.
2) /api/users/ routes to /api/Users.py:UsersApi
3) UsersApi goes to ApiServiceHandler somehow(?)
4) ApiServiceHandler.get() calls process()
5) process() calls user_authentication() because the request has auth_headers
6) the local user object is returned to process()
7) In line 30 of ApiServiceHandler, method() is called (no idea where this goes)
8) Somehow we end up in Users.py:UsersApi:get_hook which calls get_user (even though we already have the local user which was returned from user_authentication).
9) Then get_user (which already has the user object) tries to compare the user object against an empty args[0].

```
class UsersApi(ApiServiceHandler):
    """REST API handler to allow interaction with user settings"""

    def get_hook(self, user, *args):
        """Get user settings for a user"""
        # return user settings for a user based on email from access token
        return get_user(user, args[0])

    def get_user(user, user_id):
        """Get user settings"""
        if user.get_id() != long(user_id):
            return WRONG_USER_RESPONSE
        return {'user-settings': user.get_user_data_dict(), 'status': "200"}
```

**Comment by ericaddison** [comment]
haha, ok! You are right about most of that stuff. I recently changed the API so that you don't have to provide a user-id in calls to `/api/users/`, because by sending the token our server should be able to figure out who you are. The code you show is a bit out of date ... no more `args[0]` in there now.

`UsersApi` _is_ an `ApiServiceHandler`, but that base class is written in such a way that I only need to provide the core login in `xxx_hook()` for a particular http request type, instead of including a lot of boilerplate code for each API handler...

Bottom line ... if you pull the new code things should work better :)

**Comment by ericaddison** [comment]
1) The FirebaseHeader is calling /api/users/ and passing the firebase token. With the intent of authenticating the current user and retrieving the local user object from ndb.
2) /api/users/ routes to /api/Users.py:UsersApi
3) UsersApi goes to ApiServiceHandler somehow **(because UsersApi ISA ApiServiceHandler)**
4) ApiServiceHandler.get() calls process()
5) process() calls user_authentication() because the request has auth_headers
6) the local user object is returned to process()
7) In line 30 of ApiServiceHandler, method() is called (no idea where this goes) **(`method` is one of the `xxx_hooks`)**
8) Somehow we end up in Users.py:UsersApi:get_hook **(because `get_hook` is the `method` passed into `process` for a `GET` request)** which calls get_user (even though we already have the local user which was returned from user_authentication). **(the `Users::get_hook()` calls a local `get_user` method that returns JSON data suitable for sending back to the client)**
9) Then get_user (which already has the user object) tries to compare the user object against an empty args[0]. **(not anymore :) )**

**Comment by ericaddison** [comment]
tests in place
